lib/iris/coords.py

```diff
@@ -1391,6 +1394,9 @@ def __eq__(self, other):
 
     # The __ne__ operator from Coord implements the not __eq__ method.
 
+    def __hash__(self):
+        return hash(id(self))
```

Comment:
I'm not sure that is the intended hash of a coordinate. The py docs state "Hashable objects which compare equal must have the same hash value.", but we have implemented an equality which contradicts this.

The problem of using anything other than id for the hash though is that our coordinates are mutable...

Essentially we are in a bit of a pickle, as coordinates are:

 * mutable
 * hashable
 * implement ``__eq__``, which compares mutable attributes

In the medium term, we may need to rethink the use of coordinates as dictionary keys.